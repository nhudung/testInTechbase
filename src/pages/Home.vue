<template>
	<div class="wrapHome">
		<v-container>
			<h1 class="title">CORONAVIRUS COVID19</h1>
			<v-row>
				<v-col cols="12" sm="4">
					<Coutries :data="listCountry" @getByCountry="getByCountry" />
				</v-col>
				<v-col cols="12" sm="8">
					<CasesInCountry :data="listcasesByCountry" />
				</v-col>
			</v-row>
		</v-container>
		<div>
			<v-dialog v-model="dialog" width="500">
				<v-card>
					<v-card-text class="style-message">
						{{ message }}
					</v-card-text>
					<v-card-actions>
						<v-spacer></v-spacer>
						<v-btn color="primary" text @click="dialog = false">
							Close
						</v-btn>
					</v-card-actions>
				</v-card>
			</v-dialog>
		</div>
	</div>
</template>

<script>
	import Coutries from "@/components/Coutries";
	import CasesInCountry from "@/components/CasesInCountry";
	import { mapActions } from "vuex";
	export default {
		components: {
			Coutries,
			CasesInCountry,
		},
		data() {
			return {
				listCountry: [],
				listcasesByCountry: [],
				dialog: false,
				message: "",
			};
		},
		created() {
			this.getListCountry();
			this.getByCountry();
		},
		methods: {
			...mapActions({
				apiGetListCountry: "getListCountry",
				apigetByCountry: "getByCountry",
			}),
			async getListCountry() {
				const response = await this.apiGetListCountry();
				if (this.failRespone(response)) return;
				this.listCountry = response.data;
			},
			async getByCountry(slug) {
				if (!slug) {
					slug = "vietnam";
				}
				const response = await this.apigetByCountry(slug);
				if (this.failRespone(response)) return;
				this.listcasesByCountry = response.data;
			},
			failRespone(response) {
				if (!response.ok) {
					this.message = response.error.message;
					this.dialog = true;
					return true;
				}
			},
		},
	};
</script>

<style lang="scss">
	.wrapHome {
		font-family: "Roboto", sans-serif;
		.title {
			text-align: center;
			margin: 50px 0;
		}
	}
	.v-dialog > .v-card > .v-card__text.style-message {
		text-align: center;
		font-size: 20px;
		padding: 30px 0;
	}
</style>
