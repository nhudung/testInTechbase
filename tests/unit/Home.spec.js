import { shallow } from "@vue/test-utils";
import Home from "@/pages/Home.vue";

describe("Home.spec.js", () => {
	let cmp;

	beforeEach(() => {
		cmp = shallow(Home, {
			data: {
				listCountry: [
					{
						Country: "Viet Nam",
						CountryCode: "VN",
					},
				],
			},
		});
	});

	it("has the expected html structure", () => {
		expect(cmp.element).toMatchSnapshot();
	});
});
