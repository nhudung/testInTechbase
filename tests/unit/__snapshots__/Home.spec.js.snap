// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.spec.js has the expected html structure 1`] = `
<div
  class="wrapHome"
>
  <v-container>
    <h1
      class="title"
    >
      CORONAVIRUS COVID19
    </h1>
     
    <v-row>
      <v-col
        cols="12"
        sm="4"
      >
        <coutries-stub
          data=""
        />
      </v-col>
       
      <v-col
        cols="12"
        sm="8"
      >
        <casesincountry-stub
          data=""
        />
      </v-col>
    </v-row>
  </v-container>
   
  <div>
    <v-dialog
      width="500"
    >
      <v-card>
        <v-card-text
          class="style-message"
        >
          
					
				
        </v-card-text>
         
        <v-card-actions>
          <v-spacer />
           
          <v-btn
            color="primary"
            text=""
          >
            
						Close
					
          </v-btn>
        </v-card-actions>
      </v-card>
    </v-dialog>
  </div>
</div>
`;
